src/psyclone/domain/gocean/nodes/gocean_extract_node.py

```diff
@@ -141,66 +142,111 @@ def create_driver(self, input_list, output_list):
+            sym_table.add(Symbol(unique_local_name))
+            local_name = unique_local_name
+
+            # TODO: #644 - we need to identify arrays!!
             # Any variable used needs to be defined. We also need
             # to handle the kind property better and not rely on
             # a hard-coded value.
-            decl = DeclGen(prog, "real", [var_name], kind="8",
+            decl = DeclGen(prog, "real", [local_name], kind="8",
                            dimension=":,:", allocatable=True)
             prog.add(decl)
             is_input = var_name in input_list
             is_output = var_name in output_list
 
             if is_input and not is_output:
                 # We only need the pre-variable, and we can read
-                # it from the file (and allocate its size)
+                # it from the file (which also allocate space for it)
                 call = CallGen(prog,
-                               "psy_data%ReadVariable(\"{0}\", {0})"
-                               .format(var_name))
+                               "{0}%ReadVariable(\"{1}\", {2})"
+                               .format(psy_data, var_name, local_name))
                 prog.add(call)
             elif is_input:
                 # Now must be input and output:
-                # First read the pre-variable
+                # First read the pre-variable (which is also allocated):
                 call = CallGen(prog,
-                               "psy_data%ReadVariable(\"{0}\", {0})"
-                               .format(var_name))
+                               "{0}%ReadVariable(\"{1}\", {2})"
+                               .format(psy_data, var_name, local_name))
                 prog.add(call)
                 # Then declare the post variable, and and read its values
                 # (ReadVariable will also allocate it)
-                decl = DeclGen(prog, "real", [var_name+post_suffix],
+                decl = DeclGen(prog, "real", [local_name+post_suffix],
                                dimension=":,:", kind="8", allocatable=True)
                 prog.add(decl)
                 call = CallGen(prog,
-                               "psy_data%ReadVariable(\"{0}\", {0})"
-                               .format(var_name+post_suffix))
+                               "{0}%ReadVariable(\"{1}{3}\", {2}{3})"
+                               .format(psy_data, var_name, local_name,
+                                       post_suffix))
                 prog.add(call)
             else:
                 # Now the variable is output only. We need to read the
-                # post variable in, and allocate the pre variable with
-                # the same size as the post
-                decl = DeclGen(prog, "real", [var_name+post_suffix],
+                # post variable in, and create and allocate a pre variable
+                # with the same size as the post
+                decl = DeclGen(prog, "real", [local_name+post_suffix],
```

Comment:
Again, we should use the SymbolTable for this name.

Reply:
After the change to the suffix, it should not be possible that duplicates names are created, but I obviously have still added these new names to the symbol table. The symbol table will raise an exception if an existing name is reused.

Reply:
Great.